From a9caf09cb7b5055b2bd45a08d8d1c9e7aaa4846c Mon Sep 17 00:00:00 2001
From: Petr Cvek <petrcvekcz@gmail.com>
Date: Thu, 19 Sep 2024 01:20:11 +0200
Subject: [PATCH 05/27] Debug CHS/LBA translation

---
 src/block.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/block.c b/src/block.c
index 1f600b85..cd78688c 100644
--- a/src/block.c
+++ b/src/block.c
@@ -82,6 +82,8 @@ get_translation(struct drive_s *drive)
     if (host_lchs_supplied(drive))
         return TRANSLATION_HOST;
     u8 type = drive->type;
+
+    //TODO qemu?
     if (CONFIG_QEMU && type == DTYPE_ATA) {
         // Emulators pass in the translation info via nvram.
         u8 translation = rtc_read(CMOS_BIOS_DISKTRANSFLAG + drive->cntl_id/4);
@@ -103,7 +105,7 @@ get_translation(struct drive_s *drive)
     if (cylinders <= 1024 && heads <= 16 && spt <= 63)
         return TRANSLATION_NONE;
     if (cylinders * heads <= 131072)
-        return TRANSLATION_LARGE;
+        return TRANSLATION_LARGE;	//pc2005 original
     return TRANSLATION_LBA;
 }
 
-- 
2.46.1

