From dd407f57ccc2d18836faffea1ddd1089964dc0a7 Mon Sep 17 00:00:00 2001
From: Petr Cvek <petrcvekcz@gmail.com>
Date: Thu, 19 Sep 2024 01:23:05 +0200
Subject: [PATCH 08/27] debug cdrom buffer

---
 src/cdrom.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/cdrom.c b/src/cdrom.c
index a77671a4..5f9b205d 100644
--- a/src/cdrom.c
+++ b/src/cdrom.c
@@ -154,6 +154,20 @@ cdrom_boot(struct drive_s *drive)
     if (ret)
         return 3;
 
+
+#if 0
+/////////
+for (unsigned xxx=0;xxx<2048;xxx++) {
+	__dprintf("%02x ", buffer[xxx]);
+	if ((xxx%16)==15) {
+		__dprintf("\n");
+	}
+}
+__dprintf("\n");
+/////////
+#endif
+
+
     // Validity checks
     if (buffer[0])
         return 4;
-- 
2.46.1

