From 7fa2a42e7a843cdea6a708ea2dbe8395d938437c Mon Sep 17 00:00:00 2001
From: Petr Cvek <petrcvekcz@gmail.com>
Date: Thu, 19 Sep 2024 01:33:24 +0200
Subject: [PATCH 21/27] Don't spam UART API debug if you debug through UART

---
 src/serial.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/serial.c b/src/serial.c
index 88349c8a..3a179b5b 100644
--- a/src/serial.c
+++ b/src/serial.c
@@ -166,7 +166,7 @@ handle_14XX(struct bregs *regs)
 void VISIBLE16
 handle_14(struct bregs *regs)
 {
-    debug_enter(regs, DEBUG_HDL_14);
+//    debug_enter(regs, DEBUG_HDL_14);
     if (! CONFIG_SERIAL) {
         handle_14XX(regs);
         return;
-- 
2.46.1

